mitosheet/dev/create_new_step.py

```diff
@@ -0,0 +1,379 @@
+            sheet_index,
+            {param_name.replace('_id', '_header')}
+        )\n"""
+        if param_type == 'List[ColumnID]':
+            id_to_header_code += f"""{param_name} = [
+            self.mito_widget.steps_manager.curr_step.column_ids.get_column_id_by_header(sheet_index, column_header)
+            for column_header in {param_name.replace('_id', '_header')}
+        ]"""
+    return id_to_header_code
+
+def get_test_util_function_code(original_step_name: str, params: Dict[str, str]) -> str:
+    
+    step_name = get_step_name(original_step_name)
+    params_parameter_code = get_params_parameter_code(params)
+    id_to_header_code = get_id_to_header_code(params)
+
+    params_pass_code = ""
+    for param in params:
+        params_pass_code += f'\'{param}\': {param},\n                    '
+    params_pass_code.strip()
+
+    return f"""
+    @check_transpiled_code_after_call
+    def {step_name}(
+            self, 
+            {params_parameter_code}
+        ) -> bool:
+
+        {id_to_header_code}
+
+        return self.mito_widget.receive_message(
+            self.mito_widget,
+            {OPEN_BRACKET}
+                'event': 'edit_event',
+                'id': get_new_id(),
+                'type': '{step_name}_edit',
+                'step_id': get_new_id(),
+                'params': {OPEN_BRACKET}
+                    {params_pass_code}
+                {CLOSE_BRACKET}
+            {CLOSE_BRACKET}
+        )
+    """
+
+def write_to_test_utility(original_step_name: str, params: Dict[str, str]) -> None:
+    path_to_test_util = './mitosheet/tests/test_utils.py'
+
+    with open(path_to_test_util, 'r') as f:
+        code = f.read()
+        code = code.replace(TEST_MARKER, f"{get_test_util_function_code(original_step_name, params)}\n{TEST_MARKER}")
+
+    with open(path_to_test_util, 'w') as f:
+        f.write(code)
+
+
+def read_params() -> Dict[str, str]:
+    param_names = input("Enter a comma seperated list of param nams: [sheet_index, column_ids, ...] ").replace(' ', '').split(',')
+    
+    params = {}
+    for param_name in param_names:
+        type = ''
+        while type not in VALID_PARAMETER_TYPES:
+            type = input(f'Select a type from {VALID_PARAMETER_TYPES} for {param_name}: ')
+        params[param_name] = type
+    
+    return params
+
+def get_test_file_code(original_step_name: str, params: Dict[str, str]) -> str:
+    step_name = get_step_name(original_step_name)
+    params_string = ", ".join(params.keys())
+
+    return f"""#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+\"\"\"
+Contains tests for {original_step_name}
+\"\"\"
+
+import pandas as pd
+import pytest
+from mitosheet.tests.test_utils import create_mito_wrapper_dfs
+
+{step_name.upper()}_TESTS = [
+    (
+        [
+            pd.DataFrame({OPEN_BRACKET}'A': [1, 2, 3]{CLOSE_BRACKET})
```

Comment:
Maybe since we're writing this template code it would be worth creating a dataframe with more dtypes so its easier to use for more comprehensive testing.